### PR TITLE
feat: create endpoints to download `trust_update`/`seed_update`

### LINKED ISSUE #167: Endpoints to download Trust and Seed updates

### PR DESCRIPTION
**Pull request by duguorong009**
## Changes
- Create endpoints to download `TrustUpdate` & `SeedUpdate` in
  - `openrank-sdk` cli
  - `sequencer` rpc
- Add custom query types for download

Resolves #167 

## Reminders
- [x] Add a prefix to PR: `feat:` (for features), `chore:` (for chores/refactoring), `fix:` (for bug fixes), or any from: https://www.conventionalcommits.org/en/v1.0.0/#summary
- [x] Assign reviewers
- [x] Link related issues
- [x] Make it draft if it is still WIP
- [x] Write unit tests for your code if possible
- [x] Do manual testing and make sure your changes don't break the rest of the codebase

### CODE REVIEW

**Comment by lazovicff** [comment]
This was possible using `get_tx` method. We need to think how to do this:
I think the simplest and most backwards compatible way is to just read a range from RocksDB.

We need to modify this function to include a`from` as a parameter:
https://github.com/openrankprotocol/openrank/blob/main/common/src/db/mod.rs#L142
`from` will be a starting key and `num_elements` will specify how many elements to fetch
Both will be optional. If `from` is None, we start from zero, if `num_elements` is zero we go all the way to the end.

**Comment by duguorong009** [comment]
> This was possible using `get_tx` method. We need to think how to do this: I think the simplest and most backwards compatible way is to just read a range from RocksDB.
> 
> We need to modify this function to include a`from` as a parameter: https://github.com/openrankprotocol/openrank/blob/main/common/src/db/mod.rs#L142 `from` will be a starting key and `num_elements` will specify how many elements to fetch Both will be optional. If `from` is None, we start from zero, if `num_elements` is zero we go all the way to the end.

I tried to implement the logic in 49f7ae9 .
Pls check it. @lazovicff 

Related to this update, I have 2 topics.
1. Can you add more description about what endpoint would be user-friendly?
Current comment is about improving the DB utility to fetch the range of TXs.
I hope you give more detail about improve of endpoints themselves.

2. I think `read_from_end` is a bit confusing name.
As you can see in the update, the function itself reads the TXs from start to end, using lexicographical order of Key.
Hence, I think it is better to rename it.

**Comment by lazovicff** [comment]
1. You should update `GetTrustUpdateQuery` and `GetTrustUpdateQuery` to include `from` and `size`, and the endpoints should use `read_from_end` function underneath

2. Yes, we should update the name to `get_range` or similar

**Comment by duguorong009** [comment]
@lazovicff 
Pls review the PR again.

I think there could be one possible issue.
Since we add the `size` param to `Query` types, user can download the multiple Trust/SeedUpdates from endpoints.
I am not sure if it is good. (especially if the contents of updates are really big)

File: common/src/query.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+impl GetResultsQuery {
+    pub fn new(request_tx_hash: tx::TxHash, start: u32, size: u32) -> Self {
+        Self { request_tx_hash, start, size }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Getters)]
+#[getset(get = "pub")]
+pub struct GetTrustUpdateQuery {
+    from: tx::TxHash,
```
Comment (lazovicff) [inline common/src/query.rs:22]: `from` should be optional too

File: openrank-sdk/src/main.rs [hunk 2]
```diff
@@ -401,6 +439,22 @@ async fn main() -> Result<(), Box<dyn Error>> {
                 write_json_to_file(&output_path, res)?;
             }
         },
+        Method::GetTrustUpdate { from, size, config_path, output_path } => {
+            let size = if size == 0 { None } else { Some(size) };
+            let res = get_trust_update(from, size, &config_path).await?;
+            println!("TrustUpdate: {:?}", res);
```
Comment (lazovicff) [inline openrank-sdk/src/main.rs:445]: leftover prints

File: sequencer/src/rpc.rs [hunk 3]
```diff
@@ -40,6 +41,16 @@ pub trait Rpc {
 
     #[method(name = "get_txs")]
     async fn get_txs(&self, keys: Vec<(String, tx::TxHash)>) -> Result<Vec<Tx>, ErrorObjectOwned>;
+
+    #[method(name = "get_trust_update")]
+    async fn get_trust_update(
```
Comment (lazovicff) [inline sequencer/src/rpc.rs:46]:
```suggestion
    async fn get_trust_updates(
```

File: sequencer/src/rpc.rs [hunk 4]
```diff
@@ -40,6 +41,16 @@ pub trait Rpc {
...
+        &self, query: GetTrustUpdateQuery,
+    ) -> Result<Vec<TrustUpdate>, ErrorObjectOwned>;
+
+    #[method(name = "get_seed_update")]
+    async fn get_seed_update(
```
Comment (lazovicff) [inline sequencer/src/rpc.rs:51]:
```suggestion
    async fn get_seed_updates(
```

File: common/src/query.rs (hunk 1 above)
Comment (duguorong009) [reply common/src/query.rs]: Done https://github.com/openrankprotocol/openrank/pull/171/commits/2bf5128a1db45c4470e86886b3f744897b080f00

File: openrank-sdk/src/main.rs (hunk 2 above)
Comment (duguorong009) [reply openrank-sdk/src/main.rs]: Done https://github.com/openrankprotocol/openrank/pull/171/commits/2bf5128a1db45c4470e86886b3f744897b080f00

File: sequencer/src/rpc.rs (hunk 3 above)
Comment (duguorong009) [reply sequencer/src/rpc.rs]: Done https://github.com/openrankprotocol/openrank/pull/171/commits/2bf5128a1db45c4470e86886b3f744897b080f00

**Review by lazovicff — APPROVED** [review]

**Pull request merged**